x-pack/plugin/rollup/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/rollup/20_unsupported_aggs.yml

```diff
@@ -0,0 +1,164 @@
+  - skip:
+      version: " - 8.4.99"
+      reason: "rollup: unsupported aggregations errors added in 8.5.0"
+
+  - do:
+      catch: bad_request
+      search:
+        index: test
+        body:
+          size: 0
+          aggs:
+            total_memory_used_histogram:
+              histogram:
+                field: total_memory_used
+                interval: 100000
+
+
+  - match: { status: 400 }
+  - match: { error.root_cause.0.type: unsupported_aggregation_on_downsampled_field }
+  - match: { error.root_cause.0.reason: "Field [total_memory_used] of type [aggregate_metric_double] is not supported for aggregation [histogram]" }
+
+---
+"Range aggregation on aggregate_metric_double field":
+  - skip:
+      version: " - 8.4.99"
+      reason: "rollup: unsupported aggregations errors added in 8.5.0"
+
+  - do:
+      catch: bad_request
+      search:
+        index: test
+        body:
+          size: 0
+          aggs:
+            tx_range:
+              range:
+                field: total_memory_used
+                ranges:
+                  -
+                    from: 0.0
+                    to: 200.0
+                  -
+                    from: 201.0
+                    to: 400.0
+                  -
+                    from: 401.0
+
+  - match: { status: 400 }
+  - match: { error.root_cause.0.type: unsupported_aggregation_on_downsampled_field }
+  - match: { error.root_cause.0.reason: "Field [total_memory_used] of type [aggregate_metric_double] is not supported for aggregation [range]" }
+
+---
+"Cardinality aggregation on aggregate_metric_double field":
+  - skip:
+      version: " - 8.4.99"
+      reason: "rollup: unsupported aggregations errors added in 8.5.0"
+
+  - do:
+      catch: bad_request
+      search:
+        index: test
+        body:
+          size: 0
+          aggs:
+            tx_cardinality:
+              cardinality:
+                field: total_memory_used
+
+  - match: { status: 400 }
+  - match: { error.root_cause.0.type: unsupported_aggregation_on_downsampled_field }
+  - match: { error.root_cause.0.reason: "Field [total_memory_used] of type [aggregate_metric_double] is not supported for aggregation [cardinality]" }
+
+---
+"Percentiles aggregation on aggregate_metric_double field":
+  - skip:
+      version: " - 8.4.99"
+      reason: "rollup: unsupported aggregations errors added in 8.5.0"
+
+  - do:
+      catch: bad_request
+      search:
+        index: test
+        body:
+          size: 0
+          aggs:
+            tx_percentile:
+              percentiles:
+                field: total_memory_used
+                percents: [90, 95, 99]
+
+  - match: { status: 400 }
+  - match: { error.root_cause.0.type: unsupported_aggregation_on_downsampled_field }
+  - match: { error.root_cause.0.reason: "Field [total_memory_used] of type [aggregate_metric_double] is not supported for aggregation [percentiles]" }
+
+---
+"Top-metrics aggregation on aggregate_metric_double field":
```

Comment:
Not sure whether out of scope but what about a test for the mixed case (as this will be extremely common in practice)? Querying both indices at the same time, resulting in one of them succeeding and the other producing a shard failure so there's partial data in the response

Reply:
Oh that is a good test I think...I am not sure what the result is in this case. I guess we hava a shard failure for the rollup index and partial results coming from the time series index...?

Reply:
Exactly

Reply:
This test is not necessary for the errors we throw if the aggregation is not supported (aggregation on field of type `agrgegate_metric_double`). In that case we don't check if the index is a time series index or not...just the field type.